### PR TITLE
Loki/Prometheus: Improve label browser performance

### PR DESCRIPTION
**Pull request by davkal**
- raise label limits
- discard facetting responses when selector was already changed again by user
- add autoEscaper to PromLabel to prevent crash when typing regex special chars

### CODE REVIEW

**Comment by davkal** [comment]
Looks like the debounce causes some test failures.

File: public/app/plugins/datasource/prometheus/components/PrometheusMetricsBrowser.tsx [hunk 1]
```diff
@@ -370,11 +370,16 @@ export class UnthemedPrometheusMetricsBrowser extends React.Component<BrowserPro
...
 
-  async fetchValues(name: string) {
+  async fetchValues(name: string, selector: string) {
     const { languageProvider } = this.props;
     this.updateLabelState(name, { loading: true }, `Fetching values for ${name}`);
     try {
       let rawValues = await languageProvider.getLabelValues(name);
+      // If selector changed, clear loading state and discard result by returning early
+      if (selector !== buildSelector(this.state.labels)) {
+        this.updateLabelState(name, { loading: false }, '');
```
Comment (Elfo404) [inline public/app/plugins/datasource/prometheus/components/PrometheusMetricsBrowser.tsx:380]: any reason to explicitly pass an empty string to the `status` param if in `updateLabelState` defaults to the same value?

```suggestion
        this.updateLabelState(name, { loading: false });
```

**Review by Elfo404 — APPROVED** [review]
Tested both for loki and prometheus and both work like a charm 🚀

**Comment by davkal** [comment]
Flaky test. Rebased to make use of #34238

**Pull request merged**